### PR TITLE
Changing fault handling to use PV prefixes from the spreadsheet instead of from the scLinac utility class

### PR DESCRIPTION
**Pull request by lisazacarias**
Derikka astutely pointed out that the racks have three different kinds of prefix formats for some reason?

### CODE REVIEW

File: backend/fault.py [hunk 1]
```diff
@@ -12,53 +11,39 @@ def __init__(self, message):
...
-        self.rank = rank
-        self.level = level
-        self.rack = rack
-        self.suffix = suffix
         self.description = description
         self.name = name
         self.okValue = float(okValue) if okValue else None
         self.faultValue = float(faultValue) if faultValue else None
 
+        self.pv: PV = PV(prefix + ":" + suffix, connection_timeout=PV_TIMEOUT)
```
Comment (Derikka) [inline backend/fault.py:23]: I'm not very familiar with this type of in line commenting (not sure what to call this actually?). What's happening here? The colon in "self.pv: " is the source of my confusion

Comment (lisazacarias) [reply backend/fault.py:23]: Oh it's a type annotation! I'm telling pycharm that self.pv is a PV (like, I could've done self.faultValue: float = ...

File: backend/fault.py [hunk 2]
```diff
@@ -12,53 +11,39 @@ def __init__(self, message):
...
+
     def __str__(self):
         return ', '.join("%s: %s" % item for item in vars(self).items())
```
Comment (Derikka) [inline backend/fault.py]: I forgot this def __str__(self) line was in our code. Do we actually use it anywhere?

File: backend/backEnd_constants.py [hunk 3]
```diff
@@ -14,29 +15,36 @@ def __init__(self, cavityNum, rackObject):
...
-                if fault.rack != self.rack.rackName:
+        self.faults: List[Fault] = []
+
+        for csvFault in CSV_FAULTS:
+
+            if csvFault["Level"] == "RACK":
+
+                # Rack A cavities don't care about faults for Rack B and vice versa
+                if csvFault["Rack"] != self.rack.rackName:
                     continue
```
Comment (Derikka) [inline backend/backEnd_constants.py:27]: Python question: Does "continue" take us out of the current if statement, the above if statement, or does it just take us out of the current iteration in the for loop?

File: backend/backEnd_constants.py [hunk 4]
```diff
@@ -14,29 +15,36 @@ def __init__(self, cavityNum, rackObject):
...
-                prefix = self.pvPrefix
-            else:
-                prefix = self.cryomodule.pvPrefix
 
-            self.faultPVs.append((fault, PV(prefix + fault.suffix,
-                                            connection_timeout=timeout)))
+            prefix = csvFault["PV Prefix"].format(LINAC=self.linac.name,
+                                                  CRYOMODULE=self.cryomodule.name,
+                                                  RACK=self.rack.rackName,
+                                                  CAVITY=self.number)
```
Comment (Derikka) [inline backend/backEnd_constants.py:34]: How does this handle a "PV Prefix" that doesn't have a RACK string? Will it break or be fine? (This comes down to another python question and my lack of understanding on how .format works)

File: backend/backEnd_constants.py [hunk 5]
```diff
@@ -14,29 +15,36 @@ def __init__(self, cavityNum, rackObject):
...
+
+            self.faults.append(Fault(tlc=csvFault["Severity"],
+                                     severity=csvFault["Severity"],
```
Comment (Derikka) [inline backend/backEnd_constants.py]: Should `tlc=csvFault["Severity"]` be `tlc=csvFault["Three Letter Code"]`?

**Review by Derikka — COMMENTED** [review]
I only noticed one line that I thought might be a bug when reading in the tlc for a fault from the csv file; other than that, most of my comments are just questions to clarify my own understanding of the new code. Looks awesome Lisa!

File: backend/fault.py (hunk 1 above)
Comment (Derikka) [reply backend/fault.py:23]: Ah, okay! That makes sense:D Is this type of type annotation specific to pycharm or will any editor/compiler be able to read this as well? 

(I suppose we'll find out quickly as soon as we run it in production if the OPIs can or cannot)

Comment (lisazacarias) [reply backend/fault.py:23]: They should be able to run without compilation issues, it's just _only_ useful in IDEs haha So the OPIs won't get any benefit, but they won't have any problems. Also it just helps for readability even if it doesn't do anything, so I think it's a win/win

File: backend/fault.py (hunk 2 above)
Comment (lisazacarias) [reply backend/fault.py]: Nope we don't use it, I can delete it!

File: backend/backEnd_constants.py (hunk 3 above)
Comment (lisazacarias) [reply backend/backEnd_constants.py:27]: It takes us to the next iteration of the for loop! I can add a comment

File: backend/backEnd_constants.py (hunk 4 above)
Comment (lisazacarias) [reply backend/backEnd_constants.py:34]: I tested that in the python console! It just ignores them!

File: backend/backEnd_constants.py (hunk 5 above)
Comment (lisazacarias) [reply backend/backEnd_constants.py]: OH MY GOD A BUG THANK YOU

**Review by Derikka — APPROVED** [review]

**Pull request merged**